netty/src/test/java/io/grpc/netty/NettyServerTest.java

```diff
@@ -239,8 +241,13 @@ public void serverShutdown() {
         shutdownCompleted.set(null);
       }
     });
+
     assertThat(((InetSocketAddress) ns.getListenSocketAddress()).getPort()).isGreaterThan(0);
 
+    Socket socket = new Socket();
```

Comment:
This works, but is a roundabout way of making sure the event loop has completed the runnable. Would it be better to Create a single EventLoop and pass that as the boss+worker EventLoopGroup to the NettyServer constructor? Then you can submit() your own Runnable and await its completion. We could use that same EventLoop for all the tests, really, if you'd prefer a `@BeforeClass`/`@AfterClass` to manage its lifetime.

Reply:
This roundabout is just copied from the test above it. Actually `ns.getListenSocketStats()` is not required to be called in the event loop. I'm not sure submit() is the most correct way to do it.

Reply:
The one above is actually checking the worker channel options. So creating a connection appropriate and necessary there.

I agree `ns.getListenSocketStats()` isn't required to be done in the event loop, but I wasn't suggesting to call it from the event loop. I was suggesting:
```
eventLoop.submit(new Runnable() {
    @Override public void run() {}
}).await();
InternalInstrumented<SocketStats> listenSocket = ns.getListenSocketStats();
```

It is just a synchronization barrier.

Reply:
Done with fixed event loop pool.